addons/l10n_sa_invoice/views/view_move_form.xml

```diff
@@ -0,0 +1,18 @@
+<odoo>
+    <data>
+        <record id="view_move_form" model="ir.ui.view">
+            <field name="name">account.move.deliver_date</field>
+            <field name="model">account.move</field>
+            <field name="inherit_id" ref="account.view_move_form"/>
+            <field name="arch" type="xml">
+                <field name="invoice_date" position="after">
+                    <field name="l10n_sa_show_delivery_date" invisible="1"/>
+                    <field name="l10n_sa_delivery_date" attrs="{'invisible': [('l10n_sa_show_delivery_date', '=', False)], 'required': [('l10n_sa_show_delivery_date', '=', True)]}"/>
+                </field>
+                <field name="invoice_date" position="attributes">
```

Comment:
@bakboka This adaptation of the field is tricky.  Somehow, this disables the editing of the bill date on the vendor bill which is very important to be able to do.  I would honestly just remove this and check in the validation of the invoice if the date is set.  Another way is to put a more precise xpath.  


![Screenshot from 2021-11-12 09-29-36](https://user-images.githubusercontent.com/3832699/141435505-583ae3d3-b3bb-4f8a-975b-81d15412d481.png)